dev/io.openliberty.data.internal.persistence/test/io/openliberty/data/internal/persistence/service/GenerateEntityClassBytesTest.java

```diff
@@ -0,0 +1,163 @@
+import io.openliberty.data.internal.persistence.models.Empty;
+import io.openliberty.data.internal.persistence.models.Generic;
+import io.openliberty.data.internal.persistence.models.GenericInfer;
+import io.openliberty.data.internal.persistence.models.Naming;
+import io.openliberty.data.internal.persistence.models.Primitive;
+import io.openliberty.data.internal.persistence.models.Special;
+
+/**
+ * Unit tests for Record Transformer
+ */
+public class GenerateEntityClassBytesTest {
+
+    @Test
+    public void testPrimitive() throws Exception {
+        byte[] classBytes = RecordTransformer.generateEntityClassBytes(Primitive.class, Primitive.class.getName() + "Entity");
+        assertClassBytes(classBytes);
+        assertClassSignature(Primitive.class, classBytes);
+    }
+
+    @Test
+    public void testBoxed() throws Exception {
+        byte[] classBytes = RecordTransformer.generateEntityClassBytes(Boxed.class, Boxed.class.getName() + "Entity");
+        assertClassBytes(classBytes);
+        assertClassSignature(Boxed.class, classBytes);
+    }
+
+    @Test
+    public void testSpecial() throws Exception {
+        byte[] classBytes = RecordTransformer.generateEntityClassBytes(Special.class, Special.class.getName() + "Entity");
+        assertClassBytes(classBytes);
+        assertClassSignature(Special.class, classBytes);
+    }
+
+    @Test
+    public void testEmpty() throws Exception {
+        byte[] classBytes = RecordTransformer.generateEntityClassBytes(Empty.class, Empty.class.getName() + "Entity");
+        assertClassBytes(classBytes);
+        assertClassSignature(Empty.class, classBytes);
+    }
+
+    @Test
+    public void testNaming() throws Exception {
+        byte[] classBytes = RecordTransformer.generateEntityClassBytes(Naming.class, Naming.class.getName() + "Entity");
+        assertClassBytes(classBytes);
+        assertClassSignature(Naming.class, classBytes);
+    }
+
+    @Test
+    public void testGeneric() throws Exception {
+        byte[] classBytes = RecordTransformer.generateEntityClassBytes(Generic.class, Generic.class.getName() + "Entity");
+        assertClassBytes(classBytes);
+        assertClassSignature(Generic.class, classBytes);
+    }
+
+    @Test
+    public void testGenericInfer() throws Exception {
+        try {
+            RecordTransformer.generateEntityClassBytes(GenericInfer.class, GenericInfer.class.getName() + "Entity");
+            fail("Should not have created a entity class for: " + GenericInfer.class.getName());
+        } catch (IllegalStateException e) {
+            //expected
+        }
+    }
+
+    @Test
+    public void testAmbiguous() throws Exception {
+        try {
+            RecordTransformer.generateEntityClassBytes(Ambiguous.class, Ambiguous.class.getName() + "Entity");
+            fail("Should not have been able to create a well-formed entity class for: " + Ambiguous.class.getCanonicalName());
+        } catch (IllegalStateException e) {
```

Comment:
If we change the exceptions above...
```suggestion
        } catch (DataException e) {
            //expected
        }
    }

    @Test
    public void testAmbiguous() throws Exception {
        try {
            RecordTransformer.generateEntityClassBytes(Ambiguous.class, Ambiguous.class.getName() + "Entity");
            fail("Should not have been able to create a well-formed entity class for: " + Ambiguous.class.getCanonicalName());
        } catch (DataException e) {
```